modules/@apostrophecms/seo-fields-doc-type/index.js

```diff
@@ -42,6 +44,42 @@ module.exports = {
           }
         }
       };
+
+      if (self.options.seoCanonicalTypes
+        && Array.isArray(self.options.seoCanonicalTypes)
+        && self.options.seoCanonicalTypes.length) {
+
+        const req = options.apos.task.getReq();
+
+        self.options.seoCanonicalTypes.forEach(canonicalType => {
+          const name = canonicalType.startsWith('@apostrophecms/') ? canonicalType.split('@apostrophecms/')[1] : canonicalType;
+          const fieldName = `_seoCanonical${_.capitalize(_.camelCase(name))}`;
+          const moduleName = options.apos.modules[canonicalType] && options.apos.modules[canonicalType].label
```

Comment:
Can be simplified:
```javascript
const { label = name } = options.apos.modules[canonicalType] || {}
```
Also shouldn't we tell developer that the type doesn't exists in project types?

Reply:
Ok, thanks.
It is a schema, so it will validated by Apos, and there will be a message in case it does not exist.